src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -219,6 +219,8 @@ private async Task ProcessChangesAsync(ChangeFeedProcessorContext context, IRead
                 ChangeFeedProcessorUserException userException = new ChangeFeedProcessorUserException(result.Exception, context);
                 await this._healthMonitor.OnErrorAsync(context.LeaseToken, userException);
             }
+            // Prevent the change feed lease from being checkpointed if cancellation was requested
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Can you please add tests to cover the scenario? This cancellationToken is related to the call to StopAsync, is the Function shutdown tied to that? 
Should we throw a different error so the message is clear to on the user logs?

Reply:
Why are we cancelling at this point? The user Function successfully processed the changes, why not checkpoint? It would cause an unnecessary duplication of the batch?

Reply:
After the discussion on the linked issue, it is clear.